commit c21960992f66b02e223faeee625d8c3d0ef05823 (refs/changes/68/5468/1)
Author: Trent Mick <trentm@gmail.com>
Date:   2019-02-01T16:32:44-08:00 (8 months ago)
    
    TRITON-1184 add 'make cutarelease' to vm-agent and net-agent to facilitate tagging new vers

diff --git a/Makefile b/Makefile
index 00b2941..a591666 100644
--- a/Makefile
+++ b/Makefile
@@ -134,6 +134,18 @@ dumpvar:
 	fi
 	@echo "$(VAR) is '$($(VAR))'"
 
+# Here "cutting a release" is just tagging the current commit with
+# "v(package.json version)". We don't publish this to npm.
+.PHONY: cutarelease
+cutarelease:
+	@echo "# Ensure working copy is clean."
+	[[ -z `git status --short` ]]  # If this fails, the working dir is dirty.
+	@echo "# Ensure have 'json' tool."
+	which json 2>/dev/null 1>/dev/null
+	ver=$(shell cat package.json | json version) && \
+	    git tag "v$$ver" && \
+	    git push origin "v$$ver"
+
 include ./tools/mk/Makefile.deps
 ifeq ($(shell uname -s),SunOS)
 include ./tools/mk/Makefile.node_prebuilt.targ
